### PR TITLE
Fix dynamic grouping for disjoint group datasets

### PR DESCRIPTION
**Pull request by benjaminpkane**
Fixes dynamic groups in the context of a sparse group dataset. In the below example, the non-default `right` slice should have pagination for the `third` and `fourth` groups. Adds e2e and includes new `DynamicGroup` and `DynamicGroupPagination` poms.

```py
import fiftyone as fo

dataset = fo.Dataset("sparse-dynamic-groups")
dataset.persistent = True

first = fo.Group()
second = fo.Group()
third = fo.Group()
fourth = fo.Group()

one = fo.Sample(
    filepath="one.png",
    group=first.element("left"),
    scene="a",
    frame=0,
)
two = fo.Sample(
    filepath="two.png",
    group=second.element("left"),
    scene="a",
    frame=1,
)
three = fo.Sample(
    filepath="three.png",
    group=third.element("right"),
    scene="b",
    frame=0,
)
four = fo.Sample(
    filepath="four.png",
    group=fourth.element("right"),
    scene="b",
    frame=1,
)

dataset.add_samples([one, two, three, four])
session = fo.launch_app(dataset.group_by("scene", order_by="frame"))
```

https://github.com/voxel51/fiftyone/assets/19821840/c25819ce-638f-4a15-bb9d-f3791fcf3a09

### CODE REVIEW

File: e2e-pw/src/oss/poms/action-row/dynamic-group.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      eventUtils,
+      "dynamic-group-action-group-by"
+    );
+    this.orderBy = new SelectorPom(
+      page,
+      eventUtils,
+      "dynamic-group-action-order-by"
+    );
+    this.submitBtn = this.page.getByTestId("dynamic-group-action-submit");
+    this.resetBtn = this.page.getByTestId("dynamic-group-action-reset");
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/action-row/dynamic-group.ts:26]: 👍

File: e2e-pw/src/oss/poms/modal/dynamic-group-pagination-bar.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Locator, Page, expect } from "src/oss/fixtures";
+import { ModalPom } from ".";
+
+export class DynamicGroupPaginationPom {
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/modal/dynamic-group-pagination-bar.ts:4]: 🥇

File: e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  const groupByRefresh = grid.getWaitForGridRefreshPromise();
+  await grid.actionsRow.toggleCreateDynamicGroups();
+  await grid.actionsRow.groupBy("scene", "frame");
+  await groupByRefresh;
+  await grid.assert.isEntryCountTextEqualTo("1 group with slice");
+  await grid.openFirstSample();
+  await modal.sidebar.toggleSidebarGroup("GROUP");
+  await modal.sidebar.assert.verifySidebarEntryText("group.name", "left");
+  await modal.sidebar.assert.verifySidebarEntryText("scene", "a");
+  await modal.sidebar.assert.verifySidebarEntryText("frame", "0");
```
Comment (manivoxel51) [inline e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts:71]: nit: combined these into a single call to reduce lines here

File: e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  await modal.group.dynamicGroupPagination.assert.verifyPage(2);
+  await modal.group.dynamicGroupPagination.assert.verifyTooltip(1, "frame: 0");
+  await modal.group.dynamicGroupPagination.assert.verifyTooltip(2, "frame: 1");
```
Comment (manivoxel51) [inline e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts:74]: nit: same ^

**Review by manivoxel51 — APPROVED** [review]
nice Pom additions! 🍨

File: e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts (hunk 3 above)
Comment (benjaminpkane) [reply e2e-pw/src/oss/specs/groups/sparse-dynamic-groups.spec.ts:71]: You are proposing a `verifySidebarEntryTexts` method, correct? I can add

**Pull request merged**